test/fw/ptl/utils/pbs_snaputils.py

```diff
@@ -0,0 +1,2361 @@
+                args["additional_hosts"] is not None:
+            arg_no_spaces = "".join((args["additional_hosts"]).split())
+            self.all_hosts.extend(arg_no_spaces.split(","))
+        if "map_file" in arg_names and args["map_file"] is not None:
+            self.mapfile = args["map_file"]
+        if "anonymize" in arg_names and args["anonymize"] is not None:
+            self.anonymize = args["anonymize"]
+            # Create a PBSAnonymizer object
+            del_attrs = [ATTR_v, ATTR_mailfrom, ATTR_managers, ATTR_owner,
+                         ATTR_operators, ATTR_m, ATTR_M, ATTR_aclgroup,
+                         ATTR_acluser, ATTR_aclResvgroup, ATTR_aclResvuser,
+                         ATTR_name, ATTR_jobdir]
+            obf_attrs = [ATTR_euser, ATTR_egroup, ATTR_project, ATTR_A,
+                         ATTR_g]
+            self.anon_obj = PBSAnonymizer(attr_delete=del_attrs,
+                                          attr_val=obf_attrs)
+
+        # Initialize Server and Scheduler objects
+        self.server = self.__initialize_server(self.server_host, self.diag)
+        self.scheduler = Scheduler(server=self.server, diag=self.diag)
+
+        # Add self.server_host to the list of hosts
+        self.all_hosts.append(self.server_host)
+
+        # If output needs to be a tarball, create the tarfile name
+        # tarfile name = <output directory name>.tgz
+        self.outtar_path = self.outdir + ".tgz"
+
+    def __capture_cmd_output(self, host, out_path, cmd, grep_pbs=False):
+        """
+        Run a command on the host specified and capture its output
+
+        :param host: the host to run the command on
+        :type String
+        :param out_path: path of the output file for this command
+        :type String
+        :param cmd: The command to execute
+        :type List
+        :param grep_pbs: Should do "grep pbs" on the output of the command?
+        :type Boolean
+        """
+        self.logger.debug("capturing " + str(cmd))
+
+        if "qmgr" in cmd[0]:
+            # qmgr -c is being called
+            if self.du.is_localhost(host) is False:
+                # For remote hosts, wrap qmgr's command in quotes
+                cmd[2] = "\'" + cmd[2] + "\'"
+
+        with open(out_path, "w") as out_fd:
+            try:
+                cmd_stdout = out_fd
+                if grep_pbs is True:
+                    # Set stdout to a temp file
+                    _, fn = self.du.mkstemp()
+                    tempfd = open(fn, "w")
+                    cmd_stdout = tempfd
+                self.du.run_cmd(host, cmd=cmd, stdout=cmd_stdout)
+                if grep_pbs is True:
+                    # Now run grep on the temp file and store output
+                    cmd_path = self.du.which(exe="grep")
+                    cmd = [cmd_path, "[p]bs", fn]
+                    cmd_stdout = out_fd
+                    self.du.run_cmd(cmd=cmd, stdout=cmd_stdout)
```

Comment:
ah yes, you are right, we do copy all contents over so it doesn't matter.
The '[]' was just a trick to prevent grep from finding itself.
I'll replace it with a python function, it's better than calling the grep command. Thanks for the suggestion.